### PR TITLE
feat: Add focal loss function

### PR DESCRIPTION
**Pull request by Katarzyna67**
Add focal loss function

### CODE REVIEW

**Comment by Korred** [comment]
As discussed we will just use the keras/tf implementation instead see: https://www.tensorflow.org/api_docs/python/tf/keras/losses/CategoricalFocalCrossentropy
